flow-server/src/main/java/com/vaadin/flow/server/frontend/TaskGenerateReactFiles.java

```diff
@@ -249,7 +216,16 @@ private boolean fileAvailable(String fileName) {
     }
 
     private boolean missingServerRouteImport(String routesContent) {
-        return !SERVER_ROUTE_PATTERN.matcher(routesContent).find();
+        routesContent = StringUtil.removeComments(routesContent);
+        if (usesRouterBuilder(routesContent)) {
+            return !FALLBACK_COMPONENT_PATTERN.matcher(routesContent).find();
+        } else {
+            return !SERVER_ROUTE_PATTERN.matcher(routesContent).find();
+        }
+    }
+
+    private boolean usesRouterBuilder(String routesContent) {
+        return routesContent.contains("new RouterBuilder(");
```

Comment:
Should we just always check both matchers e.g.
```suggestion
        routesContent = StringUtil.removeComments(routesContent);
        return !FALLBACK_COMPONENT_PATTERN.matcher(routesContent).find() && !SERVER_ROUTE_PATTERN.matcher(routesContent).find();
```
Also fallback component pattern checks more than server route as it checks that there is a with method call.
Perhaps SERVER_ROUTE_PATTERN should also check that there is a `...serverSideRoutes` used in the routes file.

Reply:
Yep, no complaints against this suggestion.